packages/shared/src/useForm.ts

```diff
@@ -163,9 +171,15 @@ export function useForm<T extends object>({
   );
 
   useEffect(() => {
-    if (!triggerChange.current) return;
-    triggerChange.current = false;
-    onChange?.(form);
+    if (triggerChange.current) {
+      triggerChange.current = false;
+      onChange?.(form);
+    }
+
+    if (validationCallback.current) {
+      validationCallback.current(form);
+      validationCallback.current = undefined;
```

Comment:
why do we set the ref to undefined here? These should end up stale anyway since they're not in the dep array

Reply:
Refs are outside of the render cycle so they have no effect on the dependency array. ( as far as I know 😅 )

Reply:
TIL.. Shame that eslint hasn't been telling me this for the last 5 years 😔